### PR TITLE
Refactor naming to make namig agnostic of templates

### PR DESCRIPTION
**Pull request by sibille**
#399

### CODE REVIEW

File: code/src/Core/Naming/ReservedNameValidator.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
```
Comment (crutkas) [inline code/src/Core/Naming/ReservedNameValidator.cs]: This needs a header

File: code/src/Core/Naming/ExistingNamesValidator.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.Templates.Core;
```
Comment (crutkas) [inline code/src/Core/Naming/ExistingNamesValidator.cs]: This needs a header

File: code/src/Core/Naming/Validator.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
```
Comment (crutkas) [inline code/src/Core/Naming/Validator.cs]: This needs a header

**Comment by sibille** [comment]
I'll add them in, thanks!!

**Pull request merged**